source/common/access_log/access_log_impl.cc

```diff
@@ -162,6 +165,15 @@ bool NotHealthCheckFilter::evaluate(const RequestInfo::RequestInfo& info, const
   return !info.healthCheck();
 }
 
+HeaderFilter::HeaderFilter(const envoy::config::filter::accesslog::v2::HeaderFilter& config) {
+  header_data_.push_back(Router::ConfigUtility::HeaderData(config.header()));
```

Comment:
I know that you are using an existing utility, but now it seems that HeaderData() should move into an HTTP utility class and not have it tied to route anymore.

Reply:
I agree! Does [source/common/http/utility.h](https://github.com/envoyproxy/envoy/blob/master/source/common/http/utility.h) seem like the right place? Additionally should I move `Router::ConfigUtility::matchHeaders()` as well?

Reply:
I think that is a good place for it or even creating a http/header_utility.h/cc would work too.  Yes please move the matchHeaders() as well. 
Thank you!